plugins/ftdidmx/FtdiWidget.cpp

```diff
@@ -364,15 +372,35 @@ bool FtdiInterface::Write(const ola::DmxBuffer& data) {
   }
 }
 
-bool FtdiInterface::Read(unsigned char *buff, int size) {
-  int read = ftdi_read_data(&m_handle, buff, size);
-  if (read <= 0) {
+
+bool FtdiInterface::Write(ola::io::ByteString *packet) {
+  int bytesWritten = ftdi_write_data(
+                       &m_handle,
+                       const_cast<unsigned char *>(packet->data()),
+                       packet->size());
+
+  int size = packet->size();
+  if (bytesWritten < 0) {
     OLA_WARN << m_parent->Description() << " "
              << ftdi_get_error_string(&m_handle);
     return false;
-  } else {
+  } else if (bytesWritten == static_cast<int>(packet->size())) {
     return true;
+  } else {
+    OLA_WARN << "Bytes Written: " << bytesWritten \
+             << " != Packet Size: " << size;
+    return false;
+  }
+}
+
+int FtdiInterface::Read(unsigned char *buff, int size) {
+  int read = ftdi_read_data(&m_handle, buff, size);
+  OLA_DEBUG << "ftdi_read_data() read: " << read;
+  if (read <= 0) {
```

Comment:
Is reading zero an error (in the ftdi_get_error_string sense)? What if the device just doesn't reply?

Reply:
0 is OK but no response, good catch :)
(probably left over from when I was doing WriteAndRead)

Modified